src/ape/api/compiler.py

```diff
@@ -11,6 +11,13 @@ class CompilerAPI(ABC):
     def name(self) -> str:
         ...
 
+    @property
+    @abstractmethod
+    def versions(self) -> List[str]:
```

Comment:
More I think about it, more I think this is necessary
```suggestion
    def versions(self, all_paths: List[Path]) -> List[str]:
```

To explain, this would be given all paths (not just ones to recompile) and return the set of versions necessary to compile them. This is important for driving the `compilers` key in the manifest

Reply:
Alright, I've updated the ABC and ethpm `PackageCompiler`. I'll make a PR for ape-vyper.